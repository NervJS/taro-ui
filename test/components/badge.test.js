import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtBadge from '../../.temp/components/badge/index'
import AtButton from '../../.temp/components/button/index'

describe('AtBadge Snap', () => {
  it('render AtBadge -- props value', () => {
    const component = renderToString(
      <AtBadge value='3'>
        <AtButton loading>按钮文字</AtButton>
      </AtBadge>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtBadge -- props dot', () => {
    const component = renderToString(
      <AtBadge dot>
        <AtButton loading>按钮文字</AtButton>
      </AtBadge>
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtBadge -- props maxValue', () => {
    const component = renderToString(
      <AtBadge value='10' maxValue={9}>
        <AtButton loading>按钮文字</AtButton>
      </AtBadge>
    )
    expect(component).toMatchSnapshot()
  })
})
