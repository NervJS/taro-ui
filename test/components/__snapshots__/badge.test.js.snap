// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtBadge Snap render AtBadge -- props dot 1`] = `"<div class=\\"at-badge\\"><div class=\\"at-button at-button--icon at-button--normal\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefined<div class=\\"at-button__icon\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><div class=\\"at-button__text\\">按钮文字</div></div><div class=\\"at-badge__dot\\"></div></div>"`;

exports[`AtBadge Snap render AtBadge -- props maxValue 1`] = `"<div class=\\"at-badge\\"><div class=\\"at-button at-button--icon at-button--normal\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefined<div class=\\"at-button__icon\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><div class=\\"at-button__text\\">按钮文字</div></div><div class=\\"at-badge__num\\">9+</div></div>"`;

exports[`AtBadge Snap render AtBadge -- props value 1`] = `"<div class=\\"at-badge\\"><div class=\\"at-button at-button--icon at-button--normal\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefined<div class=\\"at-button__icon\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><div class=\\"at-button__text\\">按钮文字</div></div><div class=\\"at-badge__num\\">3</div></div>"`;
